src/ship.js

```diff
@@ -0,0 +1,30 @@
+function Ship(itinerary) {
+    this.itinerary = itinerary;
+    this.previousPort = null;
+    this.currentPort = itinerary.ports[0];
+    this.currentPort.addShip(this);
+};
+
+Ship.prototype.setSail = function() {
```

Comment:
There is absolutely nothing wrong with using prototypes but I would suggest to convert this into a class as it is newer :) and it offers a little bit more.